packages/react/src/menu/root/useMenuRoot.ts

```diff
@@ -174,6 +175,25 @@ export function useMenuRoot(parameters: useMenuRoot.Parameters): useMenuRoot.Ret
         return;
       }
 
+      if (
+        nextOpen === false &&
+        event?.type === 'click' &&
+        (event as PointerEvent).pointerType === 'touch' &&
+        !allowTouchToCloseRef.current
+      ) {
+        return;
+      }
+
+      // Prevent the menu from closing on mobile devices that have a delayed click event.
+      // In some cases the menu, when tapped, will fire the focus event first and then the click event.
+      // Without this guard, the menu will close immediately after opening.
+      if (nextOpen && reason === 'focus') {
+        allowTouchToCloseRef.current = false;
+        setTimeout(() => {
+          allowTouchToCloseRef.current = true;
+        }, 300);
+      }
```

Comment:
I think this timeout should have cleanup, at least right before it's called again